service/controller/resource/instance/create_drain_old_vmss.go

```diff
@@ -0,0 +1,125 @@
+package instance
+
+import (
+	"context"
+	"fmt"
+
+	corev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/azure-operator/service/controller/internal/state"
+	"github.com/giantswarm/azure-operator/service/controller/key"
+)
+
+func (r *Resource) drainOldVMSSTransition(ctx context.Context, obj interface{}, currentState state.State) (state.State, error) {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding all drainerconfigs") // nolint: errcheck
+
+	drainerConfigs := make(map[string]corev1alpha1.DrainerConfig)
+	{
+		n := metav1.NamespaceAll
+		o := metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", key.ClusterIDLabel, key.ClusterID(cr)),
+		}
+
+		list, err := r.g8sClient.CoreV1alpha1().DrainerConfigs(n).List(o)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+
+		for _, dc := range list.Items {
+			drainerConfigs[dc.Name] = dc
+		}
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d drainerconfigs", len(drainerConfigs))) // nolint: errcheck
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding all worker VMSS instances")                         // nolint: errcheck
+
+	allWorkerInstances, err := r.allInstances(ctx, cr, key.LegacyWorkerVMSSName)
+	if IsScaleSetNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find the scale set '%s'", key.LegacyWorkerVMSSName(cr))) // nolint: errcheck
+	} else if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d worker VMSS instances", len(allWorkerInstances))) // nolint: errcheck
```

Comment:
no need for `nolint: errcheck` anymore

Reply:
OK this PR is already huge, I will not remove all those.